### PR TITLE
feat: install button with dropdown for npm, pnpm, yarn and bun

### PR DESCRIPTION
**Pull request by mathisdev7**
This PR modifies a documentation component to enhance the install button functionality. The dropdown now includes `npm`, `pnpm`, `yarn`, and `bun`, allowing users to easily copy the installation command for their preferred package manager. This update improves the flexibility and user experience for developers consulting the documentation.

**Status and versioning classification:**

### CODE REVIEW

File: apps/website/src/components/ui/InstallButton.tsx [hunk 1]
```diff
@@ -1,33 +1,83 @@
...
-			type="button"
-		>
-			<span className="font-semibold text-blurple">{'>'}</span> npm install discord.js{' '}
+		<div className={`relative inline-block ${className}`}>
+			<button
+				className="cursor-pointer rounded-md border border-neutral-300 bg-white px-4 py-2 font-mono hover:bg-neutral-200 dark:border-neutral-700 dark:bg-transparent dark:hover:bg-neutral-800"
+				onClick={() => setShowDropdown(!showDropdown)}
+				type="button"
+			>
+				<span className="font-semibold text-blurple">{'>'}</span> npm install discord.js
```
Comment (vladfrangu) [inline apps/website/src/components/ui/InstallButton.tsx:30]: This makes it look really weird, `npm` is always shown but dropdown select makes it copy a diff package manager

**Comment by mathisdev7** [comment]
Reviewed the PR, can you check if it is good please, thanks.

File: apps/website/src/components/ui/InstallButton.tsx [hunk 2]
```diff
@@ -4,30 +4,110 @@ import { Copy, CopyCheck } from 'lucide-react';
...
+		case 'pnpm':
+			return 'pnpm install discord.js';
+		case 'yarn':
+			return 'yarn add discord.js';
+		case 'bun':
+			return 'bun add discord.js';
+		default:
+			return 'npm install discord.js';
+	}
+};
```
Comment (almeidx) [inline apps/website/src/components/ui/InstallButton.tsx:18]: You could just make an array of `{ packageManager: string; command: string }` or something like that, and then loop over that to generate the list items

File: apps/website/src/components/ui/InstallButton.tsx [hunk 3]
```diff
@@ -4,30 +4,110 @@ import { Copy, CopyCheck } from 'lucide-react';
...
+				{copiedText && interacted ? (
+					<CopyCheck aria-hidden size={20} className="ml-2 inline-block text-green-500" />
+				) : (
+					<Copy aria-hidden size={20} className="ml-2 inline-block" />
+				)}
+			</button>
+
+			{showDropdown && (
+				<div className="absolute left-0 mt-2 w-full rounded-md border border-neutral-300 bg-white shadow-lg dark:border-neutral-700 dark:bg-neutral-800">
+					<ul className="py-1">
```
Comment (almeidx) [inline apps/website/src/components/ui/InstallButton.tsx:55]: Maybe use [Menu from react-aria-components](https://react-spectrum.adobe.com/react-aria/Menu.html) instead

**Review by almeidx — CHANGES_REQUESTED** [review]
From a usability standpoint, doesn't make much sense that for npm, I have to open the select and then select npm (which was already selected) for it to copy the command. Perhaps we should rethink the way this works.

We could keep the button as it was before to copy the command, and have a menu to the side which selects the package manager?

**Pull request closed without merging**